src/inference/dream/estimator.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var util = require('../../util');
+var paramStruct = require('../../params/struct');
+
+// This estimator currently makes the following assumptions:
+
+// 1. The model contains exactly one `mapData`.
+
+// 2. Either a) each element of the data array is observed (passed to
+// `observe` as its second argument) exactly once in the corresponding
+// observation function, or b) each element of the data array is
+// itself an array, of which each element is observed exactly once,
+// and in the order in which they appear in the array. See below for
+// examples of each of these.
```

Comment:
don't think it has to be in the same pr necessarily but in principle: I thought that if we get observations as an object - a list of keys and values, and also extend the observe function to get an observations object and a key rather than directly the value, then we can in the first pass start every iteration with copying the object, then in every observe statement just hallucinate the value and overwrite it in the copied object and ultimately just pass this object to the second pass and it should work fine from there without additional modifications.